### DISCUSSION TITLE
*.vim in .gitignore has no effect

### DISCUSSION THREAD

**Discussion by trijezdci** [community/community#53987] [Repositories]
### Select Topic Area

Bug

### Body

I have a C project at https://github.com/m2sf/m2c which contains a vim script in https://github.com/m2sf/m2c/tree/main/ide-support that I do not want to be tracked in the project stats, so I added a .gitignore file in the top level

https://github.com/m2sf/m2c/blob/main/.gitignore

But it has no effect, the project stats include vim script.

And yes, I added the vim script after the gitignore. I even removed and added it back in several times, but to no avail, once it is added it shows up in the project stats. What's going on?

**Comment by airtower-luna** [comment]
`.gitignore` [only affects untracked files](https://git-scm.com/docs/gitignore#_description). Once you add the file, it'll be treated like any other file that's part of the repository by design.

If you mean you want it gone from the language statistics for your repo, see: https://github.com/github-linguist/linguist/blob/master/docs/overrides.md

**Reply by trijezdci** [reply]
So, how do I get such files removed from the statistics? It is silly to show in the stats since it is not part of the software itself. The stats might as well show the markdown of the README file as a language in those stats. It doesn't do that either because it makes sense that way. In the same manner, there are other kinds of auxiliary files like the .md files that shouldn't be in the stats. How do you exclude those?

**Reply by airtower-luna** [reply]
That has nothing to do with gitignore, but I just edited the answer in above (must've overlapped with you responding). That's up to Linguist configuration via `.gitattributes`. :slightly_smiling_face:
